src/axes_marker.cpp

```diff
@@ -95,7 +95,14 @@ visualization_msgs::MarkerArray GetAxesMarkerArray(
   z_axis.scale.x = z_axis.scale.y = std::max(scale * 0.1, 0.01);
   z_axis.scale.z = scale;
 
-  // Z axis orientation
+  // Z axis orientation, which has the same rotation_matrix from the given pose.
+  Eigen::Quaternionf z_quaternion(rotation_matrix);
```

Comment:
You don't need to convert back to a quaternion. Just use `pose.orientation`.

Reply:
Completed